api/matrix.ts

```diff
@@ -0,0 +1,74 @@
+import got from '../libs/got'
+import { millify } from '../libs/utils'
+import { fetchMembersCount as fetchGitterMembersCount } from './gitter'
+import { createBadgenHandler, PathArgs } from '../libs/create-badgen-handler'
+
+const BRAND_COLOR = 'black'
+
+export default createBadgenHandler({
+  title: 'Matrix',
+  examples: {
+    '/matrix/members/%23rust/matrix.org': 'members',
+    '/matrix/members/Manjaro/matrix.org': 'members',
+    '/matrix/members/thisweekinmatrix': 'members',
+    '/matrix/members/archlinux/archlinux.org': 'members',
+    '/matrix/members/redom_redom/gitter.im': 'members'
+  },
+  handlers: {
+    '/matrix/members/:room/:server?': handler
+  }
+})
+
+async function handler ({ room, server = 'matrix.org' }: PathArgs) {
+  const roomName = room.replace(/^#/, '')
+  const membersCount = await fetchMembersCount(roomName, server)
+  if (Number.isNaN(membersCount)) {
+    return {
+      subject: 'matrix',
+      status: 'unknown',
+      color: 'grey'
+    }
+  }
+
+  const status = [
+    millify(membersCount),
+    server === 'gitter.im' ? 'gitter' : '',
+    membersCount === 1 ? 'member' : 'members'
+  ].join(' ')
+
+  return {
+    subject: `#${roomName}:${server}`,
+    status,
+    color: BRAND_COLOR
+  }
+}
+
+async function fetchMembersCount(roomName: string, server: string) {
+  if (server === 'gitter.im') {
+    const [gitterOrg, gitterRoom] = roomName.split('_')
+    return fetchGitterMembersCount(gitterOrg, gitterRoom)
+  }
+  const room = await findPublicRoom(roomName, server)
+  return room?.num_joined_members
+}
+
+async function findPublicRoom(roomName: string, server: string, homeserver?: string) {
+  homeserver = homeserver || await getHomeserver(server)
+  const roomAlias = `#${roomName}:${server}`
+  const endpoint = `${homeserver}/_matrix/client/api/v1/publicRooms`
+  const searchParams = new URLSearchParams({ limit: '500' })
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const { chunk, next_batch } = await got(endpoint, { searchParams }).json<any>()
```

Comment:
Would be nice to give it a shorter timeout, for non-exists rooms. (default timeout is 1 min)